internal/backends/qt/qt_accessible.rs

```diff
@@ -0,0 +1,699 @@
+                auto data = accessible_item->data();
+                rust!(DescriptionPropertyTracker_rearm [data: Pin<&SlintAccessibleItemData> as "void*"] {
+                    data.arm_description_tracker();
+                });
+
+                auto obj = accessible_item->object();
+                auto event = QAccessibleEvent(obj, QAccessible::DescriptionChanged);
+                QAccessible::updateAccessibility(&event);
+            });
+
+        });
+    }
+}
+
+pub struct FocusDelegationPropertyTracker {
+    accessible_item: *mut c_void,
+}
+
+impl PropertyChangeHandler for FocusDelegationPropertyTracker {
+    fn notify(&self) {
+        let accessible_item = self.accessible_item;
+        cpp!(unsafe [accessible_item as "Slint_accessible_item*"] {
+            QTimer::singleShot(0, [accessible_item]() {
+                auto data = accessible_item->data();
+                rust!(FocusDelegationPropertyTracker_rearm [data: Pin<&SlintAccessibleItemData> as "void*"] {
+                    data.arm_focus_delegation_tracker();
+                });
+
+                accessible_item->delegateFocus();
+
+                return accessible_item->data();
+            });
+        });
+    }
+}
+
+#[pin_project]
+pub struct SlintAccessibleItemData {
+    #[pin]
+    state_tracker: PropertyTracker<AccessibleItemPropertiesTracker>,
+    #[pin]
+    value_tracker: PropertyTracker<ValuePropertyTracker>,
+    #[pin]
+    label_tracker: PropertyTracker<LabelPropertyTracker>,
+    #[pin]
+    description_tracker: PropertyTracker<DescriptionPropertyTracker>,
+    #[pin]
+    focus_delegation_tracker: PropertyTracker<FocusDelegationPropertyTracker>,
+    item: ItemWeak,
+}
+
+impl SlintAccessibleItemData {
+    fn new_pin_box(accessible_item: *mut c_void, item: &ItemWeak) -> Pin<Box<Self>> {
+        let state_tracker =
+            PropertyTracker::new_with_change_handler(AccessibleItemPropertiesTracker {
+                accessible_item,
+            });
+        let value_tracker =
+            PropertyTracker::new_with_change_handler(ValuePropertyTracker { accessible_item });
+        let label_tracker =
+            PropertyTracker::new_with_change_handler(LabelPropertyTracker { accessible_item });
+        let description_tracker =
+            PropertyTracker::new_with_change_handler(DescriptionPropertyTracker {
+                accessible_item,
+            });
+        let focus_delegation_tracker =
+            PropertyTracker::new_with_change_handler(FocusDelegationPropertyTracker {
+                accessible_item,
+            });
+
+        let result = Box::pin(Self {
+            state_tracker,
+            value_tracker,
+            label_tracker,
+            description_tracker,
+            focus_delegation_tracker,
+            item: item.clone(),
+        });
+
+        result.as_ref().arm_state_tracker();
+        result.as_ref().arm_value_tracker();
+        result.as_ref().arm_label_tracker();
```

Comment:
I wonder if it would perhaps make sense to communicate the type of underlying item to this function, so that we don't allocate all trackers all the time and only arm those that are needed. It seems that for example for a `Text` with `accessible-label` we end up allocating also spin box related trackers and evaluate them (even if there are no underlying dependencies, fortunately).

Does `pin_project` support enums, so that perhaps `SlintAccessibleItemData` could be an enum?

Reply:
I would love to share more code between the different implementations of the trackers, but had no good idea how to do that yet:-/

I tried hard not to hard-code which roles will trigger which properties to be queried! Not sure whether we can keep it like that, but if we change this, then I would prefer to do so in the compiler so that it we can generate errors when the wrong information is provided.

I do not get the part about the enum. Are you suggesting to have a different set of trackers based on the role or something?

Is that worth it? Olivier said these trackers are cheap.

Reply:
tracker are cheap but not free. 
Also, "armed" tracker are more expensive than unarmed ones.
(An unarmed tracker is just a few pointer large,  an armed tracker adds the dependency nodes on the heap, if any.)

Ah, also if the expression does not depends on any property (which is the case for the value of a spin box) then the tracker will never be "armed")

Reply:
> Ah, also if the expression does not depends on any property (which is the case for the value of a spin box) then the tracker will never be "armed")

Does that mean I will not get notified when the value of a spinbox changes? That actually seems to work. Is that by accident?

Reply:
actually i meant for the value of not-a-spinbox.
The spinbox's value actually depends on a property

Reply:
I consider this out of scope for this PR.